src/classes/HouseholdAccounts.cls

```diff
@@ -98,4 +157,362 @@ public inherited sharing class HouseholdAccounts {
+
+    /*******************************************************************************************************
+    * @description sets the Primary Account for each Account in the list of Accounts
+    * @param accountIds the list of Account (Id's) that need updating.
+    */
+    private static void resetPrimaryContactForAccount(List<Id>
+            accountIds, TDTM_Runnable.DmlWrapper dmlWrapper) {
+
+        // for each account, we need a set of their contacts.
+        Map<Id, Set<Id>> contactIdsByAccountId = new Map<Id, Set<Id>>();
+
+        for (Contact contactRecord : [
+                SELECT Id, AccountId, Account.npe01__One2OneContact__c
+                FROM Contact
+                WHERE AccountId IN :accountIds
+        ]) {
+            Set<Id> contactIds = contactIdsByAccountId.get(contactRecord.AccountId);
+            if (contactIds == null) {
+                contactIds = new Set<Id>();
+                contactIdsByAccountId.put(contactRecord.AccountId, contactIds);
+            }
+            contactIds.add(contactRecord.Id);
+        }
+
+        // now go thru each account and update their primary contact if the current one isn't valid
+        List<Account> accounts = new List<Account>();
+        for (Account accountRecord : [SELECT Id, npe01__One2OneContact__c FROM Account WHERE Id IN :accountIds]) {
+            Set<Id> contactIds = contactIdsByAccountId.get(accountRecord.Id);
+            if (contactIds != null && !contactIds.contains(accountRecord.npe01__One2OneContact__c)) {
+                accountRecord.npe01__One2OneContact__c = extractFirstId(contactIds);
+                accounts.add(accountRecord);
+            }
+        }
+
+        if (accounts != null && !accounts.isEmpty()) {
+            dmlWrapper.objectsToUpdate.addAll(accounts);
+        }
+    }
+
+    /*******************************************************************************************************
+    * @description returns the first Id from the specified set
+    * @param ids a set of Id's
+    * @return Id.  null if the set is empty.
+    */
+    private static Id extractFirstId(Set<Id> ids) {
+        for (Id recordId : ids) {
+            return recordId;
+        }
+
+        return null;
+    }
+
+    /*******************************************************************************************************
+    * @description for the list of AccountId's, recalc their rollups
+    * @param accountIds the list of Account (Id's) that need updating.
+    */
+    public static void rollupAccounts(List<Id> accountIds) {
+        new RLLP_OppRollup().rollupAccounts(new Set<Id>(accountIds));
+    }
+
+    /*******************************************************************************************************
+    * @description Inserts a new 1:1 or HH Account for an Individual Contact
+    * @param contactsWithoutAccounts the list of Contacts that need updating.
+    */
+    public static void createOneToOneAndHouseholdAccountsFor(List<Contact>
```

Comment:
Probably out of scope for this initial effort, but flagging this as a long running method that we should consider refactoring.

Reply:
👍🏽

Reply:
Added to design doc for further discussion!: https://salesforce.quip.com/n2vzAQOiRbT7#UMTACAjM6Uc